setup.py

```diff
@@ -235,13 +234,10 @@ def create_dir_symlink(src, dest):
         nccl_version = ".".join(str(torch.cuda.nccl.version())[:2])
     else:
         nccl_version = ".".join(map(str, torch.cuda.nccl.version()[:2]))
-    if hasattr(torch.cuda, 'is_bf16_supported'):
-        bf16_support = torch.cuda.is_bf16_supported()
 if torch_available and hasattr(torch.version, 'hip') and torch.version.hip is not None:
     hip_version = ".".join(torch.version.hip.split('.')[:2])
 torch_info = {
     "version": torch_version,
-    "bf16_support": bf16_support,
```

Comment:
as long as there is nothing relying on this field it should be safe to remove.

Reply:
In this repository, It was only used by a test, and I've replaced the usage in the test with a runtime check.

The usage in the test was only introduced 3 weeks ago in 7fc3065074885b0a2ec59c9c49e04296dd0e0fcd, so it probably hasn't been used much by other programs.

Reply:
I totally agree, but let's see what the maintainers say.

Reply:
as you pointed out the key here is that the hardware environment may chance since the installation and thus a memorized bf16 state can easily be invalid at run time. 

This would be especially true for cross-platform/cross-hardware prebuilds.

Reply:
@jeffra, also there is a new development with IPEX - which now supports BF16 on CPU! So I highly recommend to already split up into 2 different state checks - have_bf16_cpu and have_bf16_gpu as we have just went through in transformers. https://github.com/huggingface/transformers/pull/17738